packages/core/plugins/core-plugins-server-internal/src/plugin_context.ts

```diff
@@ -321,6 +321,9 @@ export function createPluginStartContext<TPlugin, TPluginDependencies>(
     uiSettings: {
       asScopedToClient: deps.uiSettings.asScopedToClient,
       globalAsScopedToClient: deps.uiSettings.globalAsScopedToClient,
+      userAsScopedToClient: deps.uiSettings.userAsScopedToClient,
+      setUserProfileSettingsClientFactoryProvider:
+        deps.uiSettings.setUserProfileSettingsClientFactoryProvider,
```

Comment:
That kind of delegation / setter APIs should be on the `setup` contract of the service/plugin. (and not being able to do so is usually a code smell by itself)

Reply:
I should be able to move it to the `setup`- I will investigate!